### PR TITLE
 Add support for multiple field group by

### PR DESCRIPTION
**Pull request by LahiruJayasekara**
- add multiple attribute group by and
                        - support user defined functions in group by
 in ballerina stream processing.

**Multiple attribute group by**
```
forever {
        from inputStream where inputStream.age > 25
        select inputStream.name, inputStream.age, count() as count
        group by inputStream.age, inputStream.school
        => (TeacherOutput [] teachers) {
            foreach t in teachers {
                outputStream.publish(t);
            }
        }
    }
```
**Using user defined group by functions**
```
function getGroupByField(int a) returns boolean {
    if (a % 2 == 0) {
        return true;
    } else {
        return false;
    }
}

forever {
        from inputStream where inputStream.age > 25
        select inputStream.name, inputStream.age, count() as count
        group by getGroupByField(inputStream.age)
        => (TeacherOutput [] teachers) {
            foreach t in teachers {
                outputStream.publish(t);
            }
        }
    }
```

## Assignees
@mohanvive , @gimantha , @grainier

### CODE REVIEW

File: tests/ballerina-unit-test/src/test/resources/test-src/streaming/streamingv2-groupby-with-functions-test.bal [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+function printTeachers(TeacherOutput e) {
+    addToGlobalEmployeeArray(e);
+}
+
+function addToGlobalEmployeeArray(TeacherOutput e) {
+    globalTeacherOutputArray[index] = e;
+    index = index + 1;
+}
```
Comment (NatashaWso2) [inline tests/ballerina-unit-test/src/test/resources/test-src/streaming/streamingv2-groupby-with-functions-test.bal]: Shall we add an empty line at the end of the file?

File: tests/ballerina-unit-test/src/test/resources/test-src/streaming/streamingv2-groupby-with-multiple-attribute-test.bal [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+function printTeachers(TeacherOutput e) {
+    addToGlobalEmployeeArray(e);
+}
+
+function addToGlobalEmployeeArray(TeacherOutput e) {
+    globalTeacherOutputArray[index] = e;
+    index = index + 1;
+}
```
Comment (NatashaWso2) [inline tests/ballerina-unit-test/src/test/resources/test-src/streaming/streamingv2-groupby-with-multiple-attribute-test.bal]: Shall we add an empty line at the end of the file?

File: tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/streaming/BallerinaStreamsV2AggregatorWithGroupByTest.java [hunk 3]
```diff
@@ -105,4 +111,57 @@ public void testSelectQueryWithAlias() {
...
+        System.setProperty("enable.siddhiRuntime", "true");
+        Assert.assertNotNull(outputTeacherEvents);
+        Assert.assertEquals(outputTeacherEvents.length, 4, "Expected events are not received");
+
+        BMap<String, BValue> teacher0 = (BMap<String, BValue>) outputTeacherEvents[0];
+        BMap<String, BValue> teacher1 = (BMap<String, BValue>) outputTeacherEvents[1];
+        BMap<String, BValue> teacher2 = (BMap<String, BValue>) outputTeacherEvents[2];
+        BMap<String, BValue> teacher3 = (BMap<String, BValue>) outputTeacherEvents[3];
+
+
```
Comment (NatashaWso2) [inline tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/streaming/BallerinaStreamsV2AggregatorWithGroupByTest.java]: Shall we remove this extra line?

File: tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/streaming/BallerinaStreamsV2AggregatorWithGroupByTest.java [hunk 4]
```diff
@@ -105,4 +111,57 @@ public void testSelectQueryWithAlias() {
...
```
Comment (NatashaWso2) [inline tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/streaming/BallerinaStreamsV2AggregatorWithGroupByTest.java]: Shall we remove this extra line?

Comment (LahiruJayasekara) [reply tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/streaming/BallerinaStreamsV2AggregatorWithGroupByTest.java]: will do the changes

File: tests/ballerina-unit-test/src/test/resources/test-src/streaming/streamingv2-groupby-with-functions-test.bal (hunk 1 above)
Comment (LahiruJayasekara) [reply tests/ballerina-unit-test/src/test/resources/test-src/streaming/streamingv2-groupby-with-functions-test.bal]: fixed

File: tests/ballerina-unit-test/src/test/resources/test-src/streaming/streamingv2-groupby-with-multiple-attribute-test.bal (hunk 2 above)
Comment (LahiruJayasekara) [reply tests/ballerina-unit-test/src/test/resources/test-src/streaming/streamingv2-groupby-with-multiple-attribute-test.bal]: fixed

File: tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/streaming/BallerinaStreamsV2AggregatorWithGroupByTest.java (hunk 3 above)
Comment (LahiruJayasekara) [reply tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/streaming/BallerinaStreamsV2AggregatorWithGroupByTest.java]: fixed

File: tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/streaming/BallerinaStreamsV2AggregatorWithGroupByTest.java (hunk 4 above)
Comment (LahiruJayasekara) [reply tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/streaming/BallerinaStreamsV2AggregatorWithGroupByTest.java]: fixed

**Pull request closed without merging**